intermediate_source/scaled_dot_product_attention_tutorial.py

```diff
@@ -0,0 +1,333 @@
+"""
+Implement High-Performance Transformers with SCALED DOT PRODUCT ATTENTION
+================================================================================
+
+"""
+
+
+######################################################################
+# Summary
+# ~~~~~~~~
+#
+# In this tutorial, we want to highlight a new ``torch.nn.functional`` function
+# that can be helpful for implementing transformer architectures. The
+# function is named ``torch.nn.functional.scaled_dot_product_attention``.
+# For detailed description of the function, see the `PyTorch documentation <https://pytorch.org/docs/master/generated/torch.nn.functional.scaled_dot_product_attention.html#torch.nn.functional.scaled_dot_product_attention>`__.
+# This function has already been incorporated into ``torch.nn.MultiheadAttention`` and ``torch.nn.TransformerEncoderLayer``.
+#
+# Overview
+# ~~~~~~~~~
+# At a high level, this PyTorch function calculates the
+# scaled dot product attention (SDPA) between query, key, and value according to
+# the definition found in the paper `Attention is all you
+# need <https://arxiv.org/abs/1706.03762>`__. While this function can be
+# written in PyTorch using existing functions, for GPU tensors this
+# function will implicitly dispatch to an optimized implementation.
```

Comment:
Since the GPU dispatch statement is made in the next section, I suggest tweaking the wording here to focus on purpose. something like:
```
While this function can be written in PyTorch using existing functions, a fused implementation can provide
large performance benefits over a naive implementation.
```